source/sass/patterns/molecules/info-bar.scss

```diff
@@ -0,0 +1,60 @@
+@import "../../variables/color";
+@import "../../variables/font";
+@import "../../mixins/positioning";
+@import "../../mixins/spacing";
+@import "../../mixins/typography";
+
+$_info-bar-gutter: 8px;
+$_info-bar-image-offset: $font-size;
+$_info-bar-image-fudge-factor: 4px;
+$_info-bars: (
+  (type: attention, background-color: $color-attention),
+  (type: info, background-color: $color-information),
+  (type: success, background-color: $color-success),
+  (type: warning, background-color: $color-warning),
+);
+
+.info-bar {
+  color: $color-text-reverse;
+  font-family: $font-secondary;
+  font-weight: 600;
+  @include set-font-size-and-line-height(scale(-1));
+  @include padding($baselinegrid-space-extra_small $_info-bar-gutter);
+  text-align: center;
+  position: relative;
+
+  @supports (display: flex) {
+    position: static;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+  }
+
+  a {
+    color: inherit;
+    text-decoration: underline;
+  }
+}
+
+.info-bar__content {
+  @supports (display: flex) {
+    display: flex;
+    align-items: center;
+  }
+}
+
+.info-bar__image {
+  @include margin($_info-bar-image-offset, inline-end);
+  @include inset($_info-bar-image-fudge-factor, block-start);
+  position: relative;
+
+  @supports (display: flex) {
+    position: static;
+  }
+}
+
+@each $info-bar in $_info-bars {
```

Comment:
Feels like too much magic...

Reply:
Seems pretty standard sass to me.

Reply:
I meant that generating class names adds complexity without really any value.

Reply:
🤔 It does mean that the class names aren't knowable to an IDE... can refactor to express explicit class names to make the developer experience more flexible.